app/src/main/java/it/niedermann/nextcloud/deck/remote/api/RequestHelper.java

```diff
@@ -1,60 +1,98 @@
 package it.niedermann.nextcloud.deck.remote.api;
 
+import android.content.Context;
+
 import androidx.annotation.NonNull;
 
-import io.reactivex.Observable;
-import io.reactivex.disposables.Disposable;
-import io.reactivex.functions.Consumer;
-import io.reactivex.plugins.RxJavaPlugins;
-import io.reactivex.schedulers.Schedulers;
+import com.nextcloud.android.sso.exceptions.NextcloudHttpRequestFailedException;
+
+import java.util.function.Supplier;
+
 import it.niedermann.nextcloud.deck.DeckLog;
+import it.niedermann.nextcloud.deck.exceptions.OfflineException;
+import it.niedermann.nextcloud.deck.remote.helpers.util.ConnectivityUtil;
 import it.niedermann.nextcloud.deck.util.ExecutorServiceProvider;
+import okhttp3.Request;
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class RequestHelper {
 
-    static {
-        RxJavaPlugins.setErrorHandler(DeckLog::logError);
+    @NonNull
+    private final ApiProvider apiProvider;
+
+    @NonNull
+    private final ConnectivityUtil connectivityUtil;
+
+    public RequestHelper(
+            @NonNull ApiProvider apiProvider,
+            @NonNull ConnectivityUtil connectivityUtil
+    ) {
```

Comment:
Hint: I rewrote the static methods of this class and added a constructor. This makes mocking easier, hope it is okay for you to have one instance around here. This also allows us to check the connectivity in one single place quite easily.
Hope this is fine for you?

Reply:
Thats fine. Just hadn't a usecase for an instance yet. Since its a singleton anyway, I just don't care.